### PR TITLE
refine as_lodtensor, test=develop

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

Function optimization
### PR changes

Others
### Describe

Refine `_as_lodtensor`, remove confusing error message and make it support list and tuple internally.
Related #25231 （to_variable support list and tuple）

### CODE REVIEW

File: python/paddle/fluid/executor.py [hunk 1]
```diff
@@ -404,29 +404,30 @@ def _as_lodtensor(data, place, dtype=None):
             >>>     ...
 
         Args:
-            data(numpy.ndarray): a instance of array
+            data(numpy.ndarray): a instance of array, scaler, list or tuple
```
Comment (chenwhql) [inline python/paddle/fluid/executor.py]: maybe the type `(numpy.ndarray)` also need to be update?

File: python/paddle/fluid/executor.py [hunk 2]
```diff
@@ -404,29 +404,30 @@ def _as_lodtensor(data, place, dtype=None):
...
+        assert dtype is not None, 'The dtype should be given when feed data is not np.ndarray'
         if np.isscalar(data):
-            assert dtype is not None, 'dtype should be given when casting python scalar to tensor'
             dtype = convert_dtype(dtype) if isinstance(
                 dtype, core.VarDesc.VarType) else dtype
             data = np.array([data]).astype(dtype)
+        elif isinstance(data, (list, tuple)):
+            dtype = convert_dtype(dtype) if isinstance(
+                dtype, core.VarDesc.VarType) else dtype
+            data = np.array(data).astype(dtype)
```
Comment (chenwhql) [inline python/paddle/fluid/executor.py]: does it need to add check here like https://github.com/PaddlePaddle/Paddle/blob/7b7e605189779e93173b650662cabb06c2d697d7/python/paddle/fluid/reader.py#L811-L819

File: python/paddle/fluid/executor.py (hunk 1 above)
Comment (zhiqiu) [reply python/paddle/fluid/executor.py]: Done, thx.

File: python/paddle/fluid/executor.py (hunk 2 above)
Comment (zhiqiu) [reply python/paddle/fluid/executor.py]: Yes, thx.

**Review by zhwesky2010 — APPROVED** [review]

**Comment by zhwesky2010** [comment]
LGTM

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**